core/src/main/resources/reference.conf

```diff
@@ -182,6 +183,33 @@ akka.persistence.r2dbc {
   }
 }
 
+// #data-partition-settings
+# Number of tables and databases that the data will be split into. The selection of data
+# partition is made from the slice of the persistenceId.
+# For example, 4 data partitions means that slice range (0 to 255) maps to data partition 0,
+# (256 to 511) to data partition 1, (512 to 767) to data partition 3, and (768 to 1023) to
+# data partition 3.
+# This configuration cannot be changed in a rolling update, since the data must be moved
+# between the tables if number of data partitions is changed.
+akka.persistence.r2dbc.data-partition {
+  # How many tables the data will be partitioned over. The tables will have
+  # the data partition as suffix, e.g. event_journal_0, event_journal_1.
+  # Must be between 1 and 1024 and a whole number divisor of 1024 (number of slices).
+  # When number-of-partitions is 1 the table name is without suffix.
+  number-of-partitions = 1
+  # How many databases the tables will be partitioned over. A database corresponds to a connection
+  # factory with its own connection pool.
+  # Must be a whole number divisor of number-of-partitions, and less than or equal to number-of-partitions.
+  # For example, number-of-partitions=8 and number-of-databases=2 means that there will be a total of
+  # 8 tables in 2 databases, i.e. 4 tables in each database.
+  # The connection-factory setting will have the data partition range as suffix, e.g. with 8 data partitions and
+  # 2 databases the connection factory settings are connection-factory-0-3, connection-factory-4-7.
+  # When number-of-databases is 1 there will only be one connection factory, without suffix.
+  # number-of-databases > 1 not supported by H2.
+  number-of-databases = 1
+}
+// #data-partition-settings
```

Comment:
@johanandren I have also implemented partitioning over multiple databases (several connection factories). Would be good if you can take a brief look before I continue with more than the journal, and other remaining tasks.

I hope these config comments explains how it's intended to work.